File: assertion-roulette/java-junit/CacheControlDirectiveType_InterfaceField_SchemaFirst.java
import static org.junit.jupiter.api.Assertions.assertEquals;
import org.junit.jupiter.api.Test;

public class CacheControlDirectiveTypeTests {

    @Test
    public void cacheControlDirectiveType_InterfaceField_SchemaFirst() {
        {
            var schema = SchemaBuilder.new().addDocumentFromString("""
                    type Query {
                        field: InterfaceType
                    }

                    interface InterfaceType {
                        field: String @cacheControl(maxAge: 500 scope: PRIVATE inheritMaxAge: true)
                    }

                    type ObjectType implements InterfaceType {
                        field: String
                    }
                    """).addDirectiveType(CacheControlDirectiveType.class).use(_ -> _).create();
            var type = schema.getType("InterfaceType", InterfaceType.class);
            var directive = type.getFields().get("field").getDirectives().stream()
                    .filter(d -> d.getType().getName().equals("cacheControl")).findSingle().orElse(null);
            var obj = directive.asValue(CacheControlDirective.class);
            assertEquals(500, obj.getMaxAge());
            assertEquals(CacheControlScope.Private, obj.getScope());
            assertEquals(true, obj.isInheritMaxAge());
        }
    }
}

File: smelly/assertion-roulette/CSharpXUnit/CSharpXUnit.Tests/Tests/CacheControlDirectiveType_InterfaceField_SchemaFirst.cs
using Xunit;

namespace DefaultNamespace
{
    public class CacheControlDirectiveTypeTests
    {
        [Fact]
        public void CacheControlDirectiveType_InterfaceField_SchemaFirst()
        {
            {
                var schema = SchemaBuilder.New().AddDocumentFromString(@"
                        type Query {
                            field: InterfaceType
                        }
            
                        interface InterfaceType {
                            field: String @cacheControl(maxAge: 500 scope: PRIVATE inheritMaxAge: true)
                        }
            
                        type ObjectType implements InterfaceType {
                            field: String
                        }
                        ").AddDirectiveType<CacheControlDirectiveType>().Use(_ => _).Create();
                var type = schema.GetType<InterfaceType>("InterfaceType");
                var directive = type.Fields["field"].Directives.Single(d => d.Type.Name == "cacheControl");
                var obj = directive.AsValue<CacheControlDirective>();
                Assert.Equal(500, obj.MaxAge);
                Assert.Equal(CacheControlScope.Private, obj.Scope);
                Assert.Equal(true, obj.InheritMaxAge);
            }
        }
    }
}
